include/deal.II/base/derivative_form.h

```diff
@@ -366,50 +370,76 @@ DerivativeForm<order, dim, spacedim, Number>::memory_consumption()
 
 
 /**
- * One of the uses of DerivativeForm is to apply it as a transformation. This
- * is what this function does.  If @p DF is DerivativeForm<1,dim,1> it computes
- * $DF * T$, if @p DF is DerivativeForm<1,dim,rank> it computes $T*DF^{T}$.
+ * One of the uses of DerivativeForm is to apply it as a linear transformation.
+ * This function returns $\nabla \mathbf F(\mathbf x) \Delta \mathbf x$, which
+ * approximates the change in $\mathbf F(\mathbf x)$ when $\mathbf x$ is changed
+ * by the amount $\Delta \mathbf x$
+ * @f[
+ *   \nabla \mathbf F(\mathbf x) \; \Delta \mathbf x
+ *   \approx
+ *   \mathbf F(\mathbf x + \Delta \mathbf x) - \mathbf F(\mathbf x).
+ * @f]
+ * The transformation corresponds to
+ * @f[
+ *   [\text{result}]_{i_1,\dots,i_k} = i\sum_{j}
+ *   \left[\nabla \mathbf F(\mathbf x)\right]_{i_1,\dots,i_k, j}
+ *   \Delta x_j
+ * @f]
+ * in index notation and corresponds to
+ * $[\Delta \mathbf x] [\nabla \mathbf F(\mathbf x)]^T$ in matrix notation.
```

Comment:
I have to add an asterisk here as well, if we decide to use asterisk for Matrix notations.

Reply:
No, let's not.